contracts/lease/src/position/mod.rs

```diff
@@ -153,16 +154,49 @@ where
         cause: Cause,
         liquidation: Coin<Asset>,
         lpn_in_assets: Price<Lpn, Asset>,
-    ) -> Option<Status<Asset>> {
+    ) -> Option<Liquidation<Asset>> {
+        match self.validate_close_amount(liquidation, lpn_in_assets) {
+            Err(ContractError::PositionCloseAmountTooSmall(_)) => None,
+            Err(ContractError::PositionCloseAmountTooBig(_)) => Some(Liquidation::Full(cause)),
+            Err(_) => unreachable!(),
+            Ok(()) => Some(Liquidation::Partial {
```

Comment:
please add an assert that the liquidation amount is less than the position amount ...